hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GPGUtils.java

```diff
@@ -128,4 +130,16 @@ public static Map<SubClusterIdInfo, Float> createUniformWeights(
     }
     return weights;
   }
+
+  @VisibleForTesting
```

Comment:
Why visible for testing? We are not testing it, right?
It'd be good to have a unit test though.

Reply:
Thanks for your suggestion! I will fix this issue.

Reply:
We are adding a configuration parameter for timeout, which is helpful for REST interfaces. In cases where a call to a ResourceManager (RM) does not return results as expected, it prevents situations where the system might get stuck.